### PR TITLE
Adding tests for M* workload built on top of JAX Stable Stack for GPUs

### PR DESCRIPTION
**Pull request by parambole**
# Description

Context: 

A) We have created a new JAX Stable Stack image for GPUs with JAX version 0.4.35.
B) We have added support for creating new M* docker images in https://github.com/AI-Hypercomputer/maxtext/pull/981.

In this PR, we are: 

1. Updating the JAX Stable Stack DAG to run Maxtext on GCP GPUs with images that are built using JAX Stable Stack.
2. mportant: Changing the notion of "stable" for Maxtext GPU images — i.e., from "Stable" to "stable stack," which now points to images built using JAX Stable Stack.

# Tests
image
image

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run one-shot tests and provided workload links above if applicable. 
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Review by RissyRan — APPROVED** [review]
LGTM! Please get @jonb377's approval for multipod's tests. Thank you!

**Review by jonb377 — APPROVED** [review]
Thanks Param!

File: dags/imagegen_devx/jax_stable_stack_e2e.py [hunk 1]
```diff
@@ -43,12 +44,24 @@
       "v4-8": [1],
       "v6e-256": [1],
   }
-  current_datetime = config.get_current_datetime()
+  train_base = (
+      "XLA_PYTHON_CLIENT_MEM_FRACTION=0.65 TF_FORCE_GPU_ALLOW_GROWTH=true "
```
Comment (bvandermoon) [inline dags/imagegen_devx/jax_stable_stack_e2e.py:48]: Just wondering what is the context for lowering XLA_PYTHON_CLIENT_MEM_FRACTION to 0.65 here?

**Review by bvandermoon — APPROVED** [review]
LGTM, thanks @parambole

File: dags/imagegen_devx/jax_stable_stack_e2e.py (hunk 1 above)
Comment (parambole) [reply dags/imagegen_devx/jax_stable_stack_e2e.py:48]: @bvandermoon I was using the same train command as in https://github.com/GoogleCloudPlatform/ml-auto-solutions/blame/master/dags/multipod/maxtext_gpu_end_to_end.py#L40.

As far as I understand, `XLA_PYTHON_CLIENT_MEM_FRACTION` specifies how much memory JAX<>XLA would have available. I am unsure what the reasoning for the original train command was, i.e., if extra memory was being reserved for operations other than JAX. But thinking about it, I believe this parameter is not required here for the tests in any case, so I have removed it.

**Pull request merged**